pareto/strategic_water_management/strategic_produced_water_optimization.py

```diff
@@ -2331,7 +2357,270 @@ def ReuseObjectiveFunctionRule(model):
         model.ReuseObjectiveFunction = Constraint(
             rule=ReuseObjectiveFunctionRule, doc="Reuse objective function"
         )
+    elif model.config.objective == Objectives.cost_surrogate:
+        if model.config.desalination_model == DesalinationModel.false:
+            raise Warning(
+                "Cannot create a surrogate objective with Desalination Model being selected"
+            )
+        from idaes.core.surrogate.surrogate_block import SurrogateBlock
+        from idaes.core.surrogate.keras_surrogate import KerasSurrogate
+
+        model.v_C_Treatment_site = Var(
+            model.s_R,
+            model.s_T,
+            initialize=0,
+            within=NonNegativeReals,
+            units=model.model_units["currency_time"],
+            doc="Cost of treating produced water at treatment site [currency]",
+        )
+        model.v_C_Treatment_site_ReLU = Var(
+            model.s_R,
+            model.s_T,
+            initialize=0,
+            within=NonNegativeReals,
+            units=model.model_units["currency_time"],
+            doc="Cost of treating produced water at treatment site with flow consideration [currency]",
+        )
+        model.v_C_TreatmentCapEx_site = Var(
+            model.s_R,
+            initialize=0,
+            within=NonNegativeReals,
+            units=model.model_units["currency"],
+            doc="Capital cost of constructing or expanding treatment capacity for each site [currency]",
+        )
+        model.v_C_TreatmentCapEx_site_time = Var(
+            model.s_R,
+            model.s_T,
+            initialize=0,
+            within=NonNegativeReals,
+            units=model.model_units["currency"],
+            doc="Capital cost of constructing or expanding treatment capacity for each time at each site [currency]",
+        )
+        model.v_C_TreatmentCapEx_surrogate = Var(
+            initialize=0,
+            within=NonNegativeReals,
+            units=model.model_units["currency"],
+            doc="Capital cost of constructing or expanding treatment capacity [currency]",
+        )
+        model.vb_y_DesalSelected = Var(
+            model.s_R,
+            within=Binary,
+            initialize=0,
+            doc="MVC selection for each desalination site",
+        )
+        model.inlet_salinity = Var(
+            model.s_R,
+            within=Reals,
+            initialize=model.df_parameters["DesalinationSurrogate"]["inlet_salinity"],
+            units=pyunits.kg / pyunits.litre,
+            doc="Inlet salinity in the feed",
+        )
+        model.recovery = Var(
+            model.s_R,
+            initialize=model.df_parameters["DesalinationSurrogate"]["recovery"],
+            within=Reals,
+            bounds=(0, 1),
+            doc="Recovery of water",
+        )
+        model.vb_y_flow_ReLU = Var(
+            model.s_R, model.s_T, initialize=0, within=Binary, doc="Binary for flow"
```

Comment:
Ok. I shall make the doc string a bit more elaborate. The variable is used as an indicator such that when the flow is higher than 0 the variable would be 1 else 0. And this would help in putting costs to 0 for 0 flow.